homeassistant/components/teslemetry/coordinator.py

```diff
@@ -65,3 +65,32 @@ def _flatten(
             else:
                 result[key] = value
         return result
+
+
+class TeslemetryEnergyDataCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Class to manage fetching data from the Teslemetry API."""
+
+    def __init__(self, hass: HomeAssistant, api: EnergySpecific) -> None:
```

Comment:
Looks like there is room to make a base class here to not have to repeat `__init__`?

Reply:
Won't that mess up the types since they accept different data objects?

Reply:
@gjohansson-ST added a base class.

Reply:
Ahh I just realised this is redundant and I will need to revert it back in a future PR, but its fine to stay as is for now.

https://github.com/Teslemetry/hass-teslemetry/blob/main/custom_components/teslemetry/coordinator.py